gengokit/httptransport/httptransport.go

```diff
@@ -239,6 +259,28 @@ func createDecodeConvertFunc(f Field) string {
 	case strings.Contains(f.GoType, "string"):
 		fType = "%s := %s"
 	}
+	if !f.IsBaseType || f.Repeated {
```

Comment:
I feel this logic might be complex enough to go into it's own function. It is pretty hard to parse right here.